Issues with loading services from database in the 4.2.1 release

I can't get a service ticket since I upgraded my CAS server from 4.2.0 to 4.2.1, I always get a NullPointerException. I have configured the jpaServiceRegistryDao.

When the RegisteredService is loaded from the database, some of the objects that are serialized as @Lob have transient fields that have not been initialized, and when they are used you get NullPointerExceptions. 

For example the field org.jasig.cas.services.DefaultRegisteredServiceAccessStrategy.logger is marked as transient in the 4.2.1 release, when hibernate deserializes an object of this class from the column "access_strategy" of the table "RegisteredServiceImpl" the value of the field logger stays as null. When the core of the CAS uses one of the methods of this object you get a NullPointerException in the first attempt to write a log entry.

This error was detected in the fields marked as @Lob in the org.jasig.cas.services.AbstractRegisteredService class, which have classes that implement the type of the field that added the keyword transient to the log fields in the 4.2.1 release. Examples: DefaultRegisteredServiceUsernameProvider, PrincipalAttributeRegisteredServiceUsernameProvider, AbstractRegisteredServiceAttributeReleasePolicy, DefaultRegisteredServiceAccessStrategy and RegisteredServicePublicKeyImpl,

This bug makes the release unusable in deployments that have the jpaServiceRegistryDao configured.

Information about the bug:
1. CAS server version: 4.2.1
2. Container type/version: Tomcat
3. Steps to duplicate the problem: Wait until the JpaServiceRegistryDaoImpl reloads the services from the database and try to login in a service
4. Stacktrace:
```
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is org.springframework.webflow.execution.ActionExecutionException: Exception thrown
 executing org.jasig.cas.web.flow.GenerateServiceTicketAction@ab5d2da in state 'generateServiceTicket' of flow 'login' -- action execution attributes were 'map[[empty]]'
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:982)
        at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:620)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:727)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
        at org.jasig.cas.security.ResponseHeadersEnforcementFilter.doFilter(ResponseHeadersEnforcementFilter.java:227)
        at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:346)
        at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:262)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
        at org.jasig.cas.security.RequestParameterPolicyEnforcementFilter.doFilter(RequestParameterPolicyEnforcementFilter.java:250)
        at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:346)
        at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:262)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
        at org.jasig.inspektr.common.web.ClientInfoThreadLocalFilter.doFilter(ClientInfoThreadLocalFilter.java:62)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
        at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:121)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:346)
        at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:262)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:503)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:170)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
        at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:950)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:421)
        at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1070)
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:611)
        at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:314)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.springframework.webflow.execution.ActionExecutionException: Exception thrown executing org.jasig.cas.web.flow.GenerateServiceTicketAction@ab5d2da in state 'generateS
erviceTicket' of flow 'login' -- action execution attributes were 'map[[empty]]'
        at org.springframework.webflow.execution.ActionExecutor.execute(ActionExecutor.java:60)
        at org.springframework.webflow.action.EvaluateAction.doExecute(EvaluateAction.java:77)
        at org.springframework.webflow.action.AbstractAction.execute(AbstractAction.java:188)
        at org.springframework.webflow.execution.AnnotatedAction.execute(AnnotatedAction.java:145)
        at org.springframework.webflow.execution.ActionExecutor.execute(ActionExecutor.java:51)
        at org.springframework.webflow.engine.ActionState.doEnter(ActionState.java:101)
        at org.springframework.webflow.engine.State.enter(State.java:194)
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228)
        at org.springframework.webflow.engine.DecisionState.doEnter(DecisionState.java:51)
        at org.springframework.webflow.engine.State.enter(State.java:194)
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228)
        at org.springframework.webflow.engine.DecisionState.doEnter(DecisionState.java:51)
        at org.springframework.webflow.engine.State.enter(State.java:194)
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.execute(FlowExecutionImpl.java:395)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.execute(RequestControlContextImpl.java:214)
        at org.springframework.webflow.engine.TransitionableState.handleEvent(TransitionableState.java:116)
        at org.springframework.webflow.engine.Flow.handleEvent(Flow.java:547)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.handleEvent(FlowExecutionImpl.java:390)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.handleEvent(RequestControlContextImpl.java:210)
        at org.springframework.webflow.engine.ActionState.doEnter(ActionState.java:105)
        at org.springframework.webflow.engine.State.enter(State.java:194)
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.execute(FlowExecutionImpl.java:395)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.execute(RequestControlContextImpl.java:214)
        at org.springframework.webflow.engine.TransitionableState.handleEvent(TransitionableState.java:116)
        at org.springframework.webflow.engine.Flow.handleEvent(Flow.java:547)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.handleEvent(FlowExecutionImpl.java:390)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.handleEvent(RequestControlContextImpl.java:210)
        at org.springframework.webflow.engine.ActionState.doEnter(ActionState.java:105)
        at org.springframework.webflow.engine.State.enter(State.java:194)
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.execute(FlowExecutionImpl.java:395)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.execute(RequestControlContextImpl.java:214)
        at org.springframework.webflow.engine.TransitionableState.handleEvent(TransitionableState.java:116)
        at org.springframework.webflow.engine.Flow.handleEvent(Flow.java:547)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.handleEvent(FlowExecutionImpl.java:390)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.handleEvent(RequestControlContextImpl.java:210)
        at org.springframework.webflow.engine.ActionState.doEnter(ActionState.java:105)
        at org.springframework.webflow.engine.State.enter(State.java:194)
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.execute(FlowExecutionImpl.java:395)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.execute(RequestControlContextImpl.java:214)
        at org.springframework.webflow.engine.TransitionableState.handleEvent(TransitionableState.java:116)
        at org.springframework.webflow.engine.Flow.handleEvent(Flow.java:547)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.handleEvent(FlowExecutionImpl.java:390)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.handleEvent(RequestControlContextImpl.java:210)
        at org.springframework.webflow.engine.ActionState.doEnter(ActionState.java:105)
        at org.springframework.webflow.engine.State.enter(State.java:194)
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.execute(FlowExecutionImpl.java:395)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.execute(RequestControlContextImpl.java:214)
        at org.springframework.webflow.engine.TransitionableState.handleEvent(TransitionableState.java:116)
        at org.springframework.webflow.engine.Flow.handleEvent(Flow.java:547)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.handleEvent(FlowExecutionImpl.java:390)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.handleEvent(RequestControlContextImpl.java:210)
        at org.springframework.webflow.engine.ActionState.doEnter(ActionState.java:105)
        at org.springframework.webflow.engine.State.enter(State.java:194)
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.execute(FlowExecutionImpl.java:395)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.execute(RequestControlContextImpl.java:214)
        at org.springframework.webflow.engine.TransitionableState.handleEvent(TransitionableState.java:116)
        at org.springframework.webflow.engine.Flow.handleEvent(Flow.java:547)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.handleEvent(FlowExecutionImpl.java:390)
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.handleEvent(RequestControlContextImpl.java:210)
        at org.springframework.webflow.engine.ActionState.doEnter(ActionState.java:105)
        at org.springframework.webflow.engine.State.enter(State.java:194)
        at org.springframework.webflow.engine.Flow.start(Flow.java:527)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.start(FlowExecutionImpl.java:368)
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.start(FlowExecutionImpl.java:223)
        at org.springframework.webflow.executor.FlowExecutorImpl.launchExecution(FlowExecutorImpl.java:140)
        at org.springframework.webflow.mvc.servlet.FlowHandlerAdapter.handle(FlowHandlerAdapter.java:238)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:959)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:893)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
        ... 43 more
Caused by: java.lang.NullPointerException
        at org.jasig.cas.services.DefaultRegisteredServiceAccessStrategy.doPrincipalAttributesAllowServiceAccess_aroundBody12(DefaultRegisteredServiceAccessStrategy.java:182)
        at org.jasig.cas.services.DefaultRegisteredServiceAccessStrategy$AjcClosure13.run(DefaultRegisteredServiceAccessStrategy.java:1)
        at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
        at org.jasig.inspektr.aspect.TraceLogAspect.traceMethod(TraceLogAspect.java:44)
        at org.jasig.cas.services.DefaultRegisteredServiceAccessStrategy.doPrincipalAttributesAllowServiceAccess(DefaultRegisteredServiceAccessStrategy.java:180)
        at org.jasig.cas.CentralAuthenticationServiceImpl.grantServiceTicket_aroundBody2(CentralAuthenticationServiceImpl.java:169)
        at org.jasig.cas.CentralAuthenticationServiceImpl$AjcClosure3.run(CentralAuthenticationServiceImpl.java:1)
        at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
        at org.jasig.inspektr.aspect.TraceLogAspect.traceMethod(TraceLogAspect.java:44)
        at org.jasig.cas.CentralAuthenticationServiceImpl.grantServiceTicket(CentralAuthenticationServiceImpl.java:136)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:302)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
        at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:85)
        at org.jasig.inspektr.audit.AuditTrailManagementAspect.handleAuditTrail(AuditTrailManagementAspect.java:128)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:621)
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:610)
        at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:68)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:168)
        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:99)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:281)
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
        at com.ryantenney.metrics.spring.MeteredMethodInterceptor.invoke(MeteredMethodInterceptor.java:45)
        at com.ryantenney.metrics.spring.MeteredMethodInterceptor.invoke(MeteredMethodInterceptor.java:32)
        at com.ryantenney.metrics.spring.AbstractMetricMethodInterceptor.invoke(AbstractMetricMethodInterceptor.java:59)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
        at com.ryantenney.metrics.spring.TimedMethodInterceptor.invoke(TimedMethodInterceptor.java:48)
        at com.ryantenney.metrics.spring.TimedMethodInterceptor.invoke(TimedMethodInterceptor.java:34)
        at com.ryantenney.metrics.spring.AbstractMetricMethodInterceptor.invoke(AbstractMetricMethodInterceptor.java:59)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
        at com.ryantenney.metrics.spring.CountedMethodInterceptor.invoke(CountedMethodInterceptor.java:46)
        at com.ryantenney.metrics.spring.CountedMethodInterceptor.invoke(CountedMethodInterceptor.java:32)
        at com.ryantenney.metrics.spring.AbstractMetricMethodInterceptor.invoke(AbstractMetricMethodInterceptor.java:59)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:208)
        at com.sun.proxy.$Proxy82.grantServiceTicket(Unknown Source)
        at org.jasig.cas.web.flow.GenerateServiceTicketAction.doExecute(GenerateServiceTicketAction.java:79)
        at org.springframework.webflow.action.AbstractAction.execute(AbstractAction.java:188)
        at org.springframework.webflow.execution.ActionExecutor.execute(ActionExecutor.java:51)
        ... 117 more

```

--- a/cas-server-core-services/src/main/java/org/jasig/cas/services/DefaultRegisteredServiceAccessStrategy.java
+++ b/cas-server-core-services/src/main/java/org/jasig/cas/services/DefaultRegisteredServiceAccessStrategy.java
@@ -36,7 +36,7 @@ public class DefaultRegisteredServiceAccessStrategy implements RegisteredService
 
     private static final long serialVersionUID = 1245279151345635245L;
 
-    protected final transient Logger logger = LoggerFactory.getLogger(this.getClass());
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultRegisteredServiceAccessStrategy.class);
 
     /** Is the service allowed at all? **/
     private boolean enabled = true;
@@ -179,30 +179,30 @@ public class DefaultRegisteredServiceAccessStrategy implements RegisteredService
     @Override
     public boolean doPrincipalAttributesAllowServiceAccess(final String principal, final Map<String, Object> principalAttributes) {
         if (this.requiredAttributes.isEmpty()) {
-            logger.debug("No required attributes are specified");
+            LOGGER.debug("No required attributes are specified");
             return true;
         }
         if (principalAttributes.isEmpty()) {
-            logger.debug("No principal attributes are found to satisfy attribute requirements");
+            LOGGER.debug("No principal attributes are found to satisfy attribute requirements");
             return false;
         }
 
         if (principalAttributes.size() < this.requiredAttributes.size()) {
-            logger.debug("The size of the principal attributes that are [{}] does not match requirements, "
+            LOGGER.debug("The size of the principal attributes that are [{}] does not match requirements, "
                     + "which means the principal is not carrying enough data to grant authorization",
                     principalAttributes);
             return false;
         }
 
         final Map<String, Set<String>> requiredAttrs = this.getRequiredAttributes();
-        logger.debug("These required attributes [{}] are examined against [{}] before service can proceed.",
+        LOGGER.debug("These required attributes [{}] are examined against [{}] before service can proceed.",
                 requiredAttrs, principalAttributes);
 
         final Sets.SetView<String> difference = Sets.intersection(requiredAttrs.keySet(), principalAttributes.keySet());
         final Set<String> copy = difference.immutableCopy();
 
         if (this.requireAllAttributes && copy.size() < this.requiredAttributes.size()) {
-            logger.debug("Not all required attributes are available to the principal");
+            LOGGER.debug("Not all required attributes are available to the principal");
             return false;
         }
 
@@ -227,18 +227,18 @@ public class DefaultRegisteredServiceAccessStrategy implements RegisteredService
             }
 
             if (!differenceInValues.isEmpty()) {
-                logger.info("Principal is authorized to access the service");
+                LOGGER.info("Principal is authorized to access the service");
                 return true;
             }
         }
-        logger.info("Principal is denied access as the required attributes for the registered service are missing");
+        LOGGER.info("Principal is denied access as the required attributes for the registered service are missing");
         return false;
     }
 
     @Override
     public boolean isServiceAccessAllowedForSso() {
         if (!this.ssoEnabled) {
-            logger.trace("Service is not authorized to participate in SSO.");
+            LOGGER.trace("Service is not authorized to participate in SSO.");
         }
         return this.ssoEnabled;
     }
@@ -246,7 +246,7 @@ public class DefaultRegisteredServiceAccessStrategy implements RegisteredService
     @Override
     public boolean isServiceAccessAllowed() {
         if (!this.enabled) {
-            logger.trace("Service is not enabled in service registry.");
+            LOGGER.trace("Service is not enabled in service registry.");
         }
 
         return this.enabled;

--- a/cas-server-core-services/src/main/java/org/jasig/cas/services/TimeBasedRegisteredServiceAccessStrategy.java
+++ b/cas-server-core-services/src/main/java/org/jasig/cas/services/TimeBasedRegisteredServiceAccessStrategy.java
@@ -4,6 +4,8 @@ import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * The {@link TimeBasedRegisteredServiceAccessStrategy} is responsible for
@@ -15,6 +17,8 @@ import org.joda.time.DateTime;
 public class TimeBasedRegisteredServiceAccessStrategy extends DefaultRegisteredServiceAccessStrategy {
 
     private static final long serialVersionUID = -6180748828025837047L;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TimeBasedRegisteredServiceAccessStrategy.class);
 
     private String startingDateTime;
 
@@ -100,7 +104,7 @@ public class TimeBasedRegisteredServiceAccessStrategy extends DefaultRegisteredS
             final DateTime st = DateTime.parse(this.startingDateTime);
 
             if (now.isBefore(st)) {
-                logger.warn("Service access not allowed because it starts at {}. Now is {}",
+                LOGGER.warn("Service access not allowed because it starts at {}. Now is {}",
                         this.startingDateTime, now);
                 return false;
             }
@@ -109,7 +113,7 @@ public class TimeBasedRegisteredServiceAccessStrategy extends DefaultRegisteredS
         if (this.endingDateTime != null) {
             final DateTime et = DateTime.parse(this.endingDateTime);
             if  (now.isAfter(et)) {
-                logger.warn("Service access not allowed because it ended at {}. Now is {}",
+                LOGGER.warn("Service access not allowed because it ended at {}. Now is {}",
                         this.endingDateTime, now);
                 return false;
             }

--- a/cas-server-integration-grouper/src/main/java/org/jasig/cas/grouper/services/GrouperRegisteredServiceAccessStrategy.java
+++ b/cas-server-integration-grouper/src/main/java/org/jasig/cas/grouper/services/GrouperRegisteredServiceAccessStrategy.java
@@ -4,6 +4,8 @@ import edu.internet2.middleware.grouperClient.api.GcGetGroups;
 import edu.internet2.middleware.grouperClient.ws.beans.WsGetGroupsResult;
 import edu.internet2.middleware.grouperClient.ws.beans.WsGroup;
 import org.jasig.cas.services.TimeBasedRegisteredServiceAccessStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
 import java.util.HashMap;
@@ -22,6 +24,8 @@ public class GrouperRegisteredServiceAccessStrategy extends TimeBasedRegisteredS
 
     private static final long serialVersionUID = -3557247044344135788L;
     private static final String GROUPER_GROUPS_ATTRIBUTE_NAME = "grouperAttributes";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GrouperRegisteredServiceAccessStrategy.class);
 
     /**
      * The enum Grouper group field.
@@ -58,30 +62,30 @@ public class GrouperRegisteredServiceAccessStrategy extends TimeBasedRegisteredS
             final GcGetGroups groupsClient = new GcGetGroups().addSubjectId(principal);
             results = groupsClient.execute().getResults();
         } catch (final Exception e) {
-            logger.warn("Grouper WS did not respond successfully. Ensure your credentials are correct "
+            LOGGER.warn("Grouper WS did not respond successfully. Ensure your credentials are correct "
                     + ", the url endpoint for Grouper WS is correctly configured and the subject " + principal
                     + "  exists in Grouper.", e);
             return false;
         }
 
         if (results == null || results.length == 0) {
-            logger.warn("Subject id [{}] could not be located. Access denied", principal);
+            LOGGER.warn("Subject id [{}] could not be located. Access denied", principal);
             return false;
         }
 
         for (final WsGetGroupsResult groupsResult : results) {
             if (groupsResult.getWsGroups() == null || groupsResult.getWsGroups().length == 0) {
-                logger.warn("No groups could be found for subject [{}]. Access denied", groupsResult.getWsSubject().getName());
+                LOGGER.warn("No groups could be found for subject [{}]. Access denied", groupsResult.getWsSubject().getName());
                 return false;
             }
 
             for (final WsGroup group : groupsResult.getWsGroups()) {
                 final String groupName = constructGrouperGroupAttribute(group);
-                logger.debug("Found group name [{}] for [{}]", groupName, principal);
+                LOGGER.debug("Found group name [{}] for [{}]", groupName, principal);
                 grouperGroups.add(groupName);
             }
         }
-        logger.debug("Adding [{}] under attribute name [{}] to collection of CAS attributes",
+        LOGGER.debug("Adding [{}] under attribute name [{}] to collection of CAS attributes",
                 grouperGroups, GROUPER_GROUPS_ATTRIBUTE_NAME);
 
         allAttributes.put(GROUPER_GROUPS_ATTRIBUTE_NAME, grouperGroups);

